src/convenience/session.ts

```diff
@@ -671,34 +787,51 @@ export class MemorySessionStorage<S> implements StorageAdapter<S> {
     readAll() {
         return this.readAllValues();
     }
-
+    /**
+     * @deprecated Use {@link keys} instead
+     */
     readAllKeys() {
-        return Array.from(this.storage.keys());
+        return Array.from(this.keys());
     }
-
+    /**
+     * @deprecated Use {@link values} instead
+     */
     readAllValues() {
-        return Array
-            .from(this.storage.keys())
-            .map((key) => this.read(key))
-            .filter((value): value is S => value !== undefined);
+        return Array.from(this.values());
     }
-
+    /**
+     * @deprecated Use {@link entries} instead
+     */
     readAllEntries() {
-        return Array.from(this.storage.keys())
-            .map((key) => [key, this.read(key)])
-            .filter((pair): pair is [string, S] => pair[1] !== undefined);
+        return Array.from(this.entries());
+    }
+
+    has(key: Key) {
+        return this.read(key) !== undefined;
     }
 
-    has(key: string) {
-        return this.storage.has(key);
+    write(key: Key, value: S) {
+        this.storage.set(cascade(key), addExpiryDate(value, this.timeToLive));
     }
 
-    write(key: string, value: S) {
-        this.storage.set(key, addExpiryDate(value, this.timeToLive));
+    delete(key: Key) {
+        this.storage.delete(cascade(key));
     }
 
-    delete(key: string) {
-        this.storage.delete(key);
+    *keys(prefix?: Key) {
+        for (const key of this.storage.keys(cascade(prefix ?? []))) yield key;
+    }
+    *entries(prefix?: Key) {
+        for (const key of this.keys(prefix)) {
+            const value = this.read(key);
```

Comment:
This is ugly and makes me wonder if the Trie should directly support timeouts. This will be useful because it can drop values while enumerating them.